include/glow/Backends/Backend.h

```diff
@@ -40,21 +40,33 @@ class Backend {
   /// Dtor.
   virtual ~Backend() = default;
 
-  /// Generate code for input function \param IR. \p placeholders is a list of
-  /// Placeholders that are mapped to the concrete input tensor for the
+  /// Generate code for input IR function \param IR. \p placeholders is a list
+  /// of Placeholders that are mapped to the concrete input tensor for the
   /// specific function.
   virtual std::unique_ptr<CompiledFunction>
   compile(std::unique_ptr<IRFunction> IR, const Context &ctx) const = 0;
 
-  /// Save the bundle for \p IR for a later standalone execution
+  /// Generate code for input function \param F. \p placeholders is a list of
+  /// Placeholders that are mapped to the concrete input tensor for the
+  /// specific function.
+  virtual std::unique_ptr<CompiledFunction> compile(Function *F,
+                                                    const Context &ctx) const {
+    auto IR = generateIR(F);
+    return compile(std::move(IR), ctx);
+  }
+
+  /// Save the bundle for \p F for a later standalone execution
   /// in \p outputDir. Make \p networkName the function name for
   /// the entry point of the network and prepend all generated
   /// files with this name.
-  virtual void save(std::unique_ptr<IRFunction> IR, llvm::StringRef outputDir,
+  virtual void save(Function *F, llvm::StringRef outputDir,
                     llvm::StringRef networkName) const {
     GLOW_UNREACHABLE("Saving a bundle is not supported by the backend");
   }
 
+  /// Generates, optimizes, and \returns IR for the given Function \p F.
+  virtual std::unique_ptr<IRFunction> generateIR(Function *F) const;
```

Comment:
Do we have users for this method?
I have mixed feeling about making this public as opposed to protected.